plugins/woocommerce/client/blocks/tests/e2e/tests/add-to-cart-with-options/add-to-cart-with-options.block_theme.spec.ts

```diff
@@ -264,11 +266,20 @@ test.describe( 'Add to Cart + Options Block', () => {
 
 		await page.goto( '/hoodie' );
 
-		const colorGreenOption = page.getByRole( 'option', {
+		let colorGreenOption = page.getByRole( 'option', {
 			name: 'Green',
 			exact: true,
 		} );
 
+		// Workaround for the template not being updated on the first load.
+		if ( ! ( await colorGreenOption.isVisible() ) ) {
+			await page.reload();
+			colorGreenOption = page.getByRole( 'option', {
+				name: 'Green',
+				exact: true,
+			} );
+		}
```

Comment:
Feel free to ignore but I think we could use `expect( ... ).toPass()` here so it's less "manual":

```suggestion
			await expect( async () => {
			        await page.reload();
			
					const colorGreenOption = page.getByRole( 'option', {
							name: 'Green',
							exact: true,
					} );
				
					await expect( colorGreenOption ).toBeEnabled();

					await page.getByLabel( 'Logo', { exact: true } ).selectOption( 'Yes' );
					await expect( colorGreenOption ).toBeDisabled();
			} ).toPass();
```

`expect( ... ).toPass()` would handle the repeating if necessary.

Reply:
I'm fine with it, but this has the issue that we reload the page in all cases instead of only reloading it when the locator doesn't match, no? I don't have a strong opinion, but the current approach feels more performant? :thinking:

Reply:
Makes sense! Let's go with current one!